sw/device/lib/base/random_order.c

```diff
@@ -5,15 +5,47 @@
 #include "sw/device/lib/base/random_order.h"
 
 #include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/macros.h"
 
-// TODO: The current implementation is just a skeleton, and currently just
-// traverses from 0 to `min_len * 2`.
+// The source of randomness for init(), which may be replaced at link-time.
+OT_WEAK
+uint32_t random_order_random_word(void) { return 0xcaffe17f; }
 
 void random_order_init(random_order_t *ctx, size_t min_len) {
-  ctx->state = 0;
-  ctx->max = min_len * 2;
+  if (min_len == 0) {
+    ctx->index = 0;
+    ctx->max = 0;
+  }
+
+  ctx->state = random_order_random_word();
+  ctx->index = 0;
+  ctx->max = bitfield_next_power_of_two32(min_len + 1);
```

Comment:
Can't tell what the intention is here but there is either an early return missing or this part can be simplified, e.g. index and max are assigned to twice.

Also, why `min_len + 1` instead of `min_len`?